### ISSUE TITLE
Randomize

### ISSUE THREAD

**Issue by mrWh1te** [mrWh1te/Botmation#83]
**Is your feature request related to a problem? Please describe.**
I want to run some BotAction's randomly, like on the flip of a coin, or roll of a die.

**Describe the solution you'd like**
BotAction that can take a number to represent the probability of the following BotAction(s) to run. ie 

```typescript
chain(
   // 12 sided die where the botaction's run, if 1 specific side is landed in the param value sided die
   runOnRoll(12)(
      // ... botaction's
   )
)
```
Maybe a helper to generate a random number?

**Describe alternatives you've considered**
Nothing

**Additional context**
Make bot's more dynamic, less static in their actions.

**Comment by mrWh1te** [comment]
introduces new "branching" BotAction `runOnDiceRoll()()` https://github.com/mrWh1te/Botmation/pull/105

**Comment by mrWh1te** [comment]
currently exploring `diceRoll()()` and `probably()()` in #105 

`diceRoll()()` limits the probability to 50% or lower with little variance
`probably()()` sets the probability directly without limitation

maybe consolidate?

**Comment by mrWh1te** [comment]
released in core v1.3
